rust/src/router/pipe_transport.rs

```diff
@@ -585,25 +575,39 @@ impl PipeTransport {
             let handlers = Arc::clone(&handlers);
             let data = Arc::clone(&data);
 
-            channel.subscribe_to_notifications(id.into(), move |notification| {
-                match serde_json::from_slice::<Notification>(notification) {
-                    Ok(notification) => match notification {
-                        Notification::SctpStateChange { sctp_state } => {
-                            data.sctp_state.lock().replace(sctp_state);
-
-                            handlers.sctp_state_change.call(|callback| {
-                                callback(sctp_state);
-                            });
-                        }
-                        Notification::Trace(trace_event_data) => {
-                            handlers.trace.call_simple(&trace_event_data);
-                        }
-                    },
-                    Err(error) => {
-                        error!("Failed to parse notification: {}", error);
+            channel.subscribe_to_fbs_notifications(
+                id.into(),
+                move |notification| match notification.event().unwrap() {
+                    notification::Event::TransportSctpStateChange => {
+                        let Ok(Some(notification::BodyRef::SctpStateChangeNotification(body))) =
+                            notification.body()
+                        else {
+                            panic!("Wrong message from worker: {notification:?}");
+                        };
+
+                        let sctp_state = SctpState::from_fbs(&body.sctp_state().unwrap());
+
+                        data.sctp_state.lock().replace(sctp_state);
+                        handlers.sctp_state_change.call(|callback| {
+                            callback(sctp_state);
+                        });
                     }
-                }
-            })
+                    notification::Event::TransportTrace => {
+                        let Ok(Some(notification::BodyRef::TransportTraceNotification(body))) =
+                            notification.body()
+                        else {
+                            panic!("Wrong message from worker: {notification:?}");
+                        };
+
+                        let trace_event_data =
+                            transport::TraceNotification::try_from(body).unwrap();
+                        handlers
+                            .trace
+                            .call_simple(&TransportTraceEventData::from_fbs(trace_event_data));
+                    }
+                    _ => unimplemented!(),
```

Comment:
Same question here. List all variants explicitly and have TODO in each of them to make sure things no longer compile when new variant is introduced. Things like this tend to eat compilation errors and cause frustration down the line.

Reply:
Same, listing all variants would mean listing variants for all other classes which are unrelated to`PipeTransport`.